### PR TITLE
add file citusdata-singlenode.yml

### PR DESCRIPTION
**Pull request by hectork01**

### CODE REVIEW

File: citusdata-singlenode.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      user_data: |-
+        export DEBIAN_FRONTEND=noninteractive
+        #Adding new source, Updating repository and Installing Packages
+        curl https://install.citusdata.com/community/deb.sh | bash
+        apt-get update -y
+        apt-get -y install postgresql-9.5-citus
+        
+        #Creating initial script to setup enviroment
+        #Include path to postgres binaries
+        cat - <<EOF >> /var/lib/postgresql/init.sh
```
Comment (szkl) [inline citusdata-singlenode.yml]: Input is missing.

File: citusdata-singlenode.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        export PATH=\$PATH:/usr/lib/postgresql/9.5/bin
+        echo "export PATH=\$PATH:/usr/lib/postgresql/9.5/bin" >> /var/lib/postgresql/.profile
```
Comment (szkl) [inline citusdata-singlenode.yml:28]: Is there any reference about this?

File: citusdata-singlenode.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        cd ~
+        mkdir -p citus/master citus/worker1 citus/worker2
+        
+        # create three normal postgres instances
+        initdb -D citus/master
+        initdb -D citus/worker1
+        initdb -D citus/worker2
+        sleep 3
```
Comment (szkl) [inline citusdata-singlenode.yml:35]: It doesn't look like needed.

File: citusdata-singlenode.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        echo "localhost 9701" >> citus/master/pg_worker_list.conf
+        echo "localhost 9702" >> citus/master/pg_worker_list.conf
+        
+        echo "shared_preload_libraries = 'citus'" >> citus/master/postgresql.conf
+        echo "shared_preload_libraries = 'citus'" >> citus/worker1/postgresql.conf
+        echo "shared_preload_libraries = 'citus'" >> citus/worker2/postgresql.conf
```
Comment (szkl) [inline citusdata-singlenode.yml]: These three commands can be folded into one like

```sh
echo "shared_preload_libraries = 'citus'" | tee --append citus/master/postgresql.conf citus/worker1/postgresql.conf citus/worker2/postgresql.conf
```

File: citusdata-singlenode.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        
+        pg_ctl -D citus/master -o "-p 9700" -l master_logfile start
+        pg_ctl -D citus/worker1 -o "-p 9701" -l worker1_logfile start
+        pg_ctl -D citus/worker2 -o "-p 9702" -l worker2_logfile start
+        sleep 3
+        psql -p 9700 -c "CREATE EXTENSION citus;"
```
Comment (szkl) [inline citusdata-singlenode.yml:43]:
```sh
for port in {9700..9702}; do psql -p $port -c "CREATE EXTENSION citus;"; done
```

**Review by szkl — CHANGES_REQUESTED** [review]

File: citusdata-singlenode.yml (hunk 5 above)
Comment (hectork01) [reply citusdata-singlenode.yml:43]: I'm having an error using that one: 
`
psql: invalid port number: "-c"
psql: invalid port number: "-c"
psql: invalid port number: "-c"
`

Comment (szkl) [reply citusdata-singlenode.yml:43]: I've updated the snippet.

Comment (hectork01) [reply citusdata-singlenode.yml:43]: Still the same error.

Comment (rjeczalik) [reply citusdata-singlenode.yml:43]: The script is executed by /bin/sh, in order to make that working you'd need to add the bash shebang:

```
      user_data: |-
         #!/bin/bash
         set -e
         ...
```

Comment (rjeczalik) [reply citusdata-singlenode.yml:43]: And it should be:

```bash
for port in {9700..9702}; do ... 
```

Without the quoting.

Comment (szkl) [reply citusdata-singlenode.yml:43]: Right.

Comment (hectork01) [reply citusdata-singlenode.yml:43]: @rjeczalik thanks, I already tried that, but still got the error:
```
server starting
server starting
server starting
psql: invalid port number: "-c"
psql: invalid port number: "-c"
psql: invalid port number: "-c"
```

However, when running the command in the terminal it works just fine:
image
@szkl maybe we should keep it as 3 lines of code instead as a loop.

Comment (szkl) [reply citusdata-singlenode.yml:43]: Did you add `#!/bin/bash` to top?

**Review by szkl — CHANGES_REQUESTED** [review]

**Comment by hectork01** [comment]
Not here, but when I test it, Yes I did add it.

File: citusdata-singlenode.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    secret_key: '${var.aws_secret_key}'
+resource:
+  aws_instance:
+    citus-single:
+      instance_type: t2.medium
+      ami: ''
+      tags:
+        Name: '${var.koding_user_username}-${var.koding_group_slug}'
+      user_data: |-
+        set -e
```
Comment (szkl) [inline citusdata-singlenode.yml:17]: ```#!/bin/bash```

**Review by szkl — CHANGES_REQUESTED** [review]

File: citusdata-singlenode.yml (hunk 6 above)
Comment (hectork01) [reply citusdata-singlenode.yml:17]: image
still the error:
image

**Review by szkl — APPROVED** [review]

**Pull request merged**